Snowflake.Data.Tests/ChunkDeserializerTest.cs

```diff
@@ -0,0 +1,80 @@
+﻿/*
+ * Copyright (c) 2021 Snowflake Computing Inc. All rights reserved.
+ */
+
+namespace Snowflake.Data.Tests
+{
+    using NUnit.Framework;
+    using Snowflake.Data.Configuration;
+    using Snowflake.Data.Core;
+    using System.IO;
+    using System.Text;
+    using System.Threading.Tasks;
+
+    [TestFixture]
+    class ChunkDeserializerTest
+    {
+        int ChunkParserVersionDefault = SFConfiguration.Instance().GetChunkParserVersion();
```

Comment:
Unless we don't have any mock library (we're on this) this kind of push/pop of the factory state makes me worried when thinking about running such tests in parallel (depending on who and how runs them). 
In such a case one test can change chunk parser delivered for the rest of the tests. 
Maybe we should indicate such tests with [NonParallelizable].
I should probably have done that within my previous PR for SFReusableChunkTest.
What do you think @sfc-gh-ext-simba-lf?

Reply:
Yes, we should have serialize v/s parallel annotation, so we can run tests in parallel as needed.

Reply:
I agree, since the parser version could affect other tests, then adding NonParallelizable should help with that

Would you like for me to add [NonParallelizable] to this test now or add it later when the tests are running in parallel

Reply:
imho lets maybe add this for now so we don't loose track on tests which rely on this

Reply:
Alright, added